packages/shared/src/hooks/referral/useReferralCampaign.ts

```diff
@@ -0,0 +1,115 @@
+import { RequestKey, generateQueryKey } from '../../lib/query';
+import AuthContext from '../../contexts/AuthContext';
+import { feature, Feature } from '../../lib/featureManagement';
+import { useFeatureIsOn } from '../../components/GrowthBookProvider';
+
+export interface ReferralCampaign {
+  referredUsersCount: number;
+  referralCountLimit: number;
+  url: string;
+}
+
+export interface UseReferralCampaign extends ReferralCampaign {
+  isCompleted: boolean;
+  isReady: boolean;
+  availableCount: number;
+  noKeysAvailable: boolean;
+}
+
+export enum ReferralCampaignKey {
+  Search = 'search',
+}
+
+export type UseReferralCampaignProps = {
+  campaignKey: ReferralCampaignKey;
+};
+
+export const campaignToReferralTargetCountMap: Record<
+  ReferralCampaignKey,
+  number
+> = { search: 5 };
+
+const campaignFeatureFlagMap: Partial<
+  Record<ReferralCampaignKey, Feature<string>>
+> = {
+  search: feature.search,
+};
+
+const useReferralCampaign = ({
+  campaignKey,
+}: UseReferralCampaignProps): UseReferralCampaign => {
+  const featureFlag = campaignFeatureFlagMap[campaignKey];
+  const isCampaignEnabled = useFeatureIsOn(featureFlag);
+  const { requestMethod } = useRequestProtocol();
+  const { user } = useContext(AuthContext);
+  const queryKey = generateQueryKey(RequestKey.ReferralCampaigns, user, {
+    referralOrigin: campaignKey,
+  });
+  const { data, isSuccess } = useQuery(
+    queryKey,
+    async () => {
+      const result = await requestMethod<{
+        referralCampaign: ReferralCampaign;
+      }>(
+        graphqlUrl,
+        REFERRAL_CAMPAIGN_QUERY,
+        { referralOrigin: campaignKey },
+        { requestKey: JSON.stringify(queryKey) },
+      );
+
+      return result.referralCampaign;
+    },
+    {
+      enabled: !!user?.id && !!isCampaignEnabled,
+    },
+  );
+  const referralCountLimit =
+    data?.referralCountLimit ?? campaignToReferralTargetCountMap[campaignKey];
+  const referredUsersCount = data?.referredUsersCount ?? 0;
+  const referralCurrentCount =
+    referredUsersCount > referralCountLimit
+      ? referralCountLimit
+      : referredUsersCount;
+
+  return {
+    referredUsersCount,
+    referralCountLimit,
+    url: data?.url,
+    isReady: isSuccess,
+    isCompleted: referralCurrentCount >= referralCountLimit,
+    availableCount: referralCountLimit - referredUsersCount,
+    noKeysAvailable: referralCountLimit - referredUsersCount <= 0,
+  };
+};
+
+interface UseFeatureCampaign {
+  referralCampaign: UseReferralCampaign;
+  canInvite: boolean;
+}
+
+const useFeatureCampaign = (
+  props: UseReferralCampaignProps,
+): UseFeatureCampaign => {
+  const referralCampaign = useReferralCampaign(props);
+  const { url } = referralCampaign;
+
+  return {
+    referralCampaign,
+    get canInvite() {
+      try {
+        const link = new URL(url);
+
+        return link.searchParams.has('token');
+      } catch (err) {
+        return false;
+      }
+    },
```

Comment:
Not sure I'm following this one correctly?

Reply:
Replaced by the property returned by API now.